python/lsst/cell_coadds/_fits.py

```diff
@@ -241,12 +259,47 @@ def readAsMultipleCellCoadd(self) -> MultipleCellCoadd:
             outer_cell_size = Extent2I(header["OCELL1"], header["OCELL2"])
             psf_image_size = Extent2I(header["PSFSIZE1"], header["PSFSIZE2"])
 
+            # Attempt to get inputs for each cell.
+            inputs = GridContainer[list[ObservationIdentifiers]](shape=grid.shape)
+            if written_version >= "0.3":
+                visit_dict = {
+                    row["visit"]: VisitRecord(
+                        visit=row["visit"],
+                        physical_filter=row["physical_filter"],
+                        day_obs=row["day_obs"],
+                    )
+                    for row in hdu_list[hdu_list.index_of("VISIT")].data
+                }
+                link_table = hdu_list[hdu_list.index_of("CELL")].data
+                for link_row in link_table:
+                    cell_id = Index2D(link_row["cell_x"], link_row["cell_y"])
+                    visit = link_row["visit"]
+                    obs_id = ObservationIdentifiers(
+                        instrument=header["INSTRUME"],
+                        packed=link_row["packed"],
+                        visit=visit,
+                        detector=link_row["detector"],
+                        day_obs=visit_dict[visit].day_obs,
+                        physical_filter=visit_dict[visit].physical_filter,
```

Comment:
I may have gone a bit too far in creating and using the `VisitRecord` dataclass to avoid any positional argument. `visit_dict` its scope only in this function and is not persisted. So I'd be okay dropping this.